skrifa/src/outline/glyf/mod.rs

```diff
@@ -827,25 +937,267 @@ impl<'a> Scaler<'a> {
+        self.phantom[0].x = bounds[0] as f32 - lsb as f32;
+        self.phantom[0].y = 0.0;
+        self.phantom[1].x = self.phantom[0].x + advance as f32;
+        self.phantom[1].y = 0.0;
         // vertical:
-        self.phantom[2].x = F26Dot6::ZERO;
-        self.phantom[2].y = F26Dot6::from_bits(bounds[3] as i32 + tsb);
-        self.phantom[3].x = F26Dot6::ZERO;
-        self.phantom[3].y = self.phantom[2].y - F26Dot6::from_bits(vadvance);
+        self.phantom[2].x = 0.0;
+        self.phantom[2].y = bounds[3] as f32 + tsb as f32;
+        self.phantom[3].x = 0.0;
+        self.phantom[3].y = self.phantom[2].y - vadvance as f32;
+    }
+
+    fn coords(&self) -> &[F2Dot14] {
+        self.coords
+    }
+
+    fn outlines(&self) -> &Outlines {
+        &self.outlines
+    }
+
+    fn load_simple(&mut self, glyph: &SimpleGlyph, glyph_id: GlyphId) -> Result<(), DrawError> {
+        use DrawError::InsufficientMemory;
+        // Compute the ranges for our point/flag buffers and slice them.
+        let points_start = self.point_count;
+        let point_count = glyph.num_points();
+        let phantom_start = point_count;
+        let points_end = points_start + point_count + PHANTOM_POINT_COUNT;
+        let point_range = points_start..points_end;
+        // Points and flags are accumulated as we load the outline.
+        let points = self
+            .memory
+            .points
+            .get_mut(point_range.clone())
+            .ok_or(InsufficientMemory)?;
+        let flags = self
+            .memory
+            .flags
+            .get_mut(point_range)
+            .ok_or(InsufficientMemory)?;
+        glyph.read_points_fast(&mut points[..point_count], &mut flags[..point_count])?;
+        // Compute the range for our contour end point buffer and slice it.
+        let contours_start = self.contour_count;
+        let contour_end_pts = glyph.end_pts_of_contours();
+        let contour_count = contour_end_pts.len();
+        let contours_end = contours_start + contour_count;
+        let contours = self
+            .memory
+            .contours
+            .get_mut(contours_start..contours_end)
+            .ok_or(InsufficientMemory)?;
+        // Read the contour end points.
+        for (end_pt, contour) in contour_end_pts.iter().zip(contours.iter_mut()) {
+            *contour = end_pt.get();
+        }
+        // Adjust the running point/contour total counts
+        self.point_count += point_count;
+        self.contour_count += contour_count;
+        // Append phantom points to the outline.
+        for (i, phantom) in self.phantom.iter().enumerate() {
+            points[phantom_start + i] = *phantom;
+            flags[phantom_start + i] = Default::default();
+        }
+        // Acquire deltas
+        if self.outlines.gvar.is_some() && !self.coords.is_empty() {
+            let gvar = self.outlines.gvar.as_ref().unwrap();
+            let glyph = deltas::SimpleGlyph {
```

Comment:
HB diff: this requires different a full floating point delta computation as well as application of deltas to points after processing each tuple